app/com/linkedin/drelephant/exceptions/ExceptionFinder.java

```diff
@@ -0,0 +1,167 @@
+package com.linkedin.drelephant.exceptions;
```

Comment:
Create an ExceptionFinder Interface and rename the current class to AzkabanExceptionFinder which will implement the ExceptionFinder interface. This will make your logic easily extendable to other scheduler exceptions.

Once you make this change, in the ```Application.java```, when the exception controller is called, you can make a query to the db to find the ```scheduler``` which triggered this flow and then instantiate the corresponding ExceptionFinder.